panic on scaleway.resourceScalewayRdbUserRead

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

1.3.7

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* scaleway_rdb_user

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "scaleway_rdb_user" "odoo" {
  instance_id = module.postgresql.instance.id
  name        = "odoo"
  password    = "odoo"
  is_admin    = false
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist (https://gist.github.com/) containing the complete debug output.
Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

Stack trace from the terraform-provider-scaleway_v2.9.1 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 36 [running]:
github.com/scaleway/terraform-provider-scaleway/v2/scaleway.resourceScalewayRdbUserRead({0x10347f028, 0x14000788120}, 0x140005bc500, {0x1032ce580?, 0x14000a92d60?})
        github.com/scaleway/terraform-provider-scaleway/v2/scaleway/resource_rdb_user.go:141 +0x5c0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1400028fce0, {0x10347f060, 0x1400061b170}, 0xd?, {0x1032ce580, 0x14000a92d60})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:724 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x1400028fce0, {0x10347f060, 0x1400061b170}, 0x1400062a9c0, {0x1032ce580, 0x14000a92d60})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x468
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x14000221c50, {0x10347f060?, 0x1400061b050?}, 0x14000618380)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x400
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14000741e00, {0x10347f060?, 0x14000a4be60?}, 0x140009ff6e0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:748 +0x3e8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x103431300?, 0x14000741e00}, {0x10347f060, 0x14000a4be60}, 0x14000a46a80, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400072e3c0, {0x103483360, 0x14000134340}, 0x14000157560, 0x14000851530, 0x103b39c30, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xb7c
google.golang.org/grpc.(*Server).handleStream(0x1400072e3c0, {0x103483360, 0x14000134340}, 0x14000157560, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x290

Error: The terraform-provider-scaleway_v2.9.1 plugin crashed!


### Expected Behavior

No crash

### Actual Behavior

Crash

### Steps to Reproduce

Just run

1. `terraform plan`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/docs/resources/rdb_user.md
+++ b/docs/resources/rdb_user.md
@@ -21,7 +21,7 @@ resource "random_password" "db_password" {
 
 resource "scaleway_rdb_user" "db_admin" {
   instance_id = scaleway_rdb_instance.main.id
-  name        = "titi"
+  name        = "devtools"
   password    = random_password.db_password.result
   is_admin    = true
 }

--- a/scaleway/resource_rdb_user.go
+++ b/scaleway/resource_rdb_user.go
@@ -3,6 +3,7 @@ package scaleway
 import (
 	"context"
 	"fmt"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
@@ -136,6 +137,11 @@ func resourceScalewayRdbUserRead(ctx context.Context, d *schema.ResourceData, me
 			return nil
 		}
 		return diag.FromErr(err)
+	}
+	if len(res.Users) == 0 {
+		tflog.Warn(ctx, fmt.Sprintf("couldn'd find user with name: [%s]", userName))
+		d.SetId("")
+		return nil
 	}
 
 	user := res.Users[0]

--- a/scaleway/resource_rdb_user.go
+++ b/scaleway/resource_rdb_user.go
@@ -3,9 +3,9 @@ package scaleway
 import (
 	"context"
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"strings"
 
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"

